src/plugins/intel_gna/src/ops/gna_max_pool.hpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <limits>
+
+#include "ngraph/node.hpp"
+#include "openvino/op/op.hpp"
+#include "openvino/op/util/attr_types.hpp"
+#include "openvino/op/util/max_pool_base.hpp"
+
+namespace ov {
+namespace intel_gna {
+namespace op {
+/// \brief Batched max pooling operation.
+class GNAMaxPool : public ov::op::Op {
```

Comment:
No, we need a GNAMaxPool operation. We substitute MaxPool with GNAMaxPool + Transposes in ngraph transformations. Then we sink these transposes to the end and start of the network. We cannot do this without GNAMaxPool

Reply:
Why not substitute Convolution and MaxPool with GNAConvolution + fused MaxPool with transposes?

Reply:
> Why not substitute Convolution and MaxPool with GNAConvolution + fused MaxPool with transposes?
Because it would not be a common implementation, and we would need to support different match pattern (Conv+Transpose+MAxPool, Conv+Activation+MaxPool, etc). Current implementation is more generic.

Reply:
I know that at this point discussion about design is futile, but just for the sake of understanding: the code may be generic in one place with this design, but still requires handling fusing and handling more cases in a different place. I we had a pass which sorts all involved operations in an order which is used by GNA HW convolution (conv + bias + act + pool) first, then there would be no need to make the transformation less generic, I guess. And this would simplify the backend a lot.

Reply:
I don't see any problem if we have different transformations, that do different pipeline steps. Now we substitute 
Convolution/MaxPool with GNAConvolution/GNAMaxPool + Transposes. Than we run transpose/gather sinking to remove transposes from the graph. Fusing is the one of the next steps. I think we could analyze and implement plugin improvements in a different PRs. I don't see any problems for the later improvements, but I hope in a separate PR.

Reply:
Discussed that offline with @sirzabek , extension of this transformation should be done in separate PR together with changes in the backend.